### PR TITLE
Improve force_close_file config based on discussion in #93

### PR DESCRIPTION
**Pull request by ruflin**
Add additional tests to check for improvement.

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -147,11 +147,16 @@ lines. The `backoff` value will be multiplied each time with the `backoff_factor
...
+the file, no new file with the same name can be created during this time.
 
-You can force Filebeat to close the file as soon as the file name changes by setting the `force_close_windows_files` option to true. This option is supported on Windows only. The default is false.
+You can force Filebeat to close the file as soon as the file name changes by setting the `force_close_windows_files` option to true.
+This option is supported on Windows only. The default is false.
 
-Turning on this option can lead to loss of data on rotated files. After file rotation, the beginning of the new file might be skipped because the reading starts at the end of the file. We recommend that you leave this option set to false, and instead specify a lower value for the `ignore_older` option to release files faster.
+Turning on this option can lead to loss of data on rotated files. After file rotation, the beginning of the
+new file might be skipped because the reading starts at the end of the file. We recommend that you leave this
+option set to false, and instead specify a lower value for the `ignore_older` option to release files faster.
```
Comment (tsg) [inline docs/configuration.asciidoc]: Since we know that a low ignore_older causes issues as well, we might want to remove this recommendation for now?

Comment (ruflin) [inline docs/configuration.asciidoc]: Good point, will update it.

Comment (ruflin) [inline docs/configuration.asciidoc]: Updated and some more improvements made to the docs so they are more accurate.

File: harvester/log.go [hunk 2]
```diff
@@ -271,10 +271,12 @@ func (h *Harvester) handleReadlineError(lastTimeRead time.Time, err error) error
...
 	}
 
-	// On windows, check if the file name exists (see #93)
 	if h.Config.ForceCloseFiles {
+		// Check if the file name exists (see #93)
 		_, statErr := os.Stat(h.file.Name())
-		if statErr != nil {
+
+		// Error means file does not exist. If no error, check if same file. If not close as rotated.
+		if statErr != nil || !input.IsSameFile(h.file.Name(), info) {
```
Comment (tsg) [inline harvester/log.go]: Hmm, now sure I get this. IsSameFile seems to use `os.Stat()` which will for sure return an error and hence always be false.

Comment (urso) [inline harvester/log.go]: it's using or. only if statErr==nil we will check IsSameFile

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -147,11 +147,13 @@ lines. The `backoff` value will be multiplied each time with the `backoff_factor
...
+By default, Filebeat keeps the files that it’s reading open until the timespan specified by `ignore_older` has elapsed.
+This behaviour can cause issues when a file is removed. On Windows, the file cannot be fully removed until Filebeat closes
+the file. In addition no new file with the same name can be created during this time.
 
-You can force Filebeat to close the file as soon as the file name changes by setting the `force_close_windows_files` option to true. This option is supported on Windows only. The default is false.
-
-Turning on this option can lead to loss of data on rotated files. After file rotation, the beginning of the new file might be skipped because the reading starts at the end of the file. We recommend that you leave this option set to false, and instead specify a lower value for the `ignore_older` option to release files faster.
+You can force Filebeat to close the file as soon as the file name changes by setting the
+`force_close_windows_files` option to true. The default is false. Turning on this option can lead to loss of data on
```
Comment (urso) [inline docs/configuration.asciidoc]: force_close_windows_files?

Comment (ruflin) [inline docs/configuration.asciidoc]: fixed

**Pull request merged**